### ISSUE TITLE
About the file cone_activity_net300.pth

### ISSUE THREAD

**Issue by enlighten0707** [Tenglon/hyperbolic_action#4]
Hi! Thanks for your great work and the shared code.
I have a question. As for the file **_cone_activity_net300.pth_**, is it the hyperbolic representation of action labels? Then how is it generated?
Looking forward to your reply. Thanks!

**Comment by Tenglon** [comment]
Yes, you are absolutely correct. Each vector in this array corresponds to the hyperbolic representation of one action label.

We use the very similar method as in
Ganea, O., Becigneul, G. &amp; Hofmann, T.. (ICML 2018). Hyperbolic Entailment Cones for Learning Hierarchical Embeddings. 

But **beware of the difference** that we update all the hyperbolic representations simultaneously (i.e. update the entire hyperbolic embedding matrix in each training step) rather than updating one vector in each training step (as in ICML18 paper). **This difference is vital for the performance.** 

I would upload the embedding training code soon. 
Thanks for your comment!

**Comment by guikunchen** [comment]
> Yes, you are absolutely correct. Each vector in this array corresponds to the hyperbolic representation of one action label.
> 
> We use the very similar method as in Ganea, O., Becigneul, G. & Hofmann, T.. (ICML 2018). Hyperbolic Entailment Cones for Learning Hierarchical Embeddings.
> 
> But **beware of the difference** that we update all the hyperbolic representations simultaneously (i.e. update the entire hyperbolic embedding matrix in each training step) rather than updating one vector in each training step (as in ICML18 paper). **This difference is vital for the performance.**
> 
> I would upload the embedding training code soon. Thanks for your comment!

@Tenglon  Hi! did you upload the embedding training code? Looking forward to your reply. Thanks!
